### PR TITLE
feat(github): support all crud operations for github-workflow jobs

### PR DESCRIPTION
**Pull request by lexfelixpost**
In my custom projects I wanted to generate standard workflows that are also dynamic based on which components is enabled. While still allowing users to overwrite these defaults in case it is not to their liking. I saw this is not possible because the job property is private and we only have add functions in the workflow class. 

I've added the getJob, updateJob(s), removeJob functions so users can do this.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mrgrain — CHANGES_REQUESTED** [review]
@lexfelixpost Some minor bits/questions. And your PR description MUST include the license footer.

```
---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
```

File: src/github/workflows.ts [hunk 1]
```diff
@@ -137,30 +137,78 @@ export class GithubWorkflow extends Component {
...
+   */
+  public removeJob(id: string) {
+    const updatedJobs = Object.entries(this.jobs).filter(
+      ([jobId, _job]: [
+        string,
+        workflows.Job | workflows.JobCallingReusableWorkflow
+      ]) => {
+        return jobId !== id;
+      }
+    );
```
Comment (mrgrain) [inline src/github/workflows.ts:209]:
```suggestion
    const updatedJobs = Object.entries(this.jobs).filter(([jobId]: [string]) => jobId !== id);
```

Comment (mrgrain) [reply src/github/workflows.ts:209]: Not sure if the type on the filter parameter `[jobId]: [string]` is needed at all here.

File: src/github/workflows.ts [hunk 2]
```diff
@@ -137,30 +137,78 @@ export class GithubWorkflow extends Component {
...
+    jobs: Record<string, workflows.Job | workflows.JobCallingReusableWorkflow>
+  ) {
+    verifyJobConstraints(jobs);
+
+    const newJobIds = Object.keys(jobs);
+    const updatedJobs = Object.entries(this.jobs).map(
+      ([jobId, job]: [
+        string,
+        workflows.Job | workflows.JobCallingReusableWorkflow
+      ]) => {
```
Comment (mrgrain) [inline src/github/workflows.ts:185]: Just wondering the type is necessary here, if so that's all good to keep

**Review by mrgrain — CHANGES_REQUESTED** [review]

File: src/github/workflows.ts (hunk 2 above)
Comment (lexfelixpost) [reply src/github/workflows.ts]: The typings are not needed removed it from here and filter function on line 198.

**Review by mrgrain — APPROVED** [review]
🚀

**Pull request merged**